File: labServiceUIJava/labServiceUIJava/src/main/java/com/example/labserviceuijava/service/ServiceConcert.java
package com.example.labserviceuijava.service;

import com.example.labserviceuijava.model.Concert;
import com.example.labserviceuijava.repository.ConcertDBRepository;
import com.example.labserviceuijava.repository.exceptions.RepositoryException;
import com.example.labserviceuijava.validator.ConcertValidator;

import java.util.Collection;

public class ServiceConcert {

    private final ConcertDBRepository concertDBRepository;

    private final ConcertValidator concertValidator;

    public ServiceConcert(ConcertDBRepository concertDBRepository, ConcertValidator concertValidator) {
        this.concertDBRepository = concertDBRepository;
        this.concertValidator = concertValidator;
    }

    public Iterable<Concert> findAll() {
        return concertDBRepository.findAll();
    }

    public Collection<Concert> getConcertsByDate(String date){
        return concertDBRepository.getConcertsByDate(date);
    }

    public void updateConcertTickets(Integer id, Integer numberOfTickets){
        Concert concert=concertDBRepository.findOne(id);
        int newTicketsAvailable=concert.getTicketsAvailable()-numberOfTickets;
        int newTicketsSold=concert.getTicketsSold()+numberOfTickets;
        concertDBRepository.updateConcertTickets(id,newTicketsAvailable,newTicketsSold);
    }

    public Concert getConcertByDateLocationStart(String date,String location,String startTime){
        return concertDBRepository.getConcertByDateLocationStart(date,location,startTime);
    }

}

File: labServiceUIcs/WindowsFormsApp2/service/ServiceConcert.cs
using System.Collections;
using System.Collections.Generic;
using WindowsFormsApp2.model;
using WindowsFormsApp2.repository;

namespace WindowsFormsApp2
{
    public class ServiceConcert
    {
        private ConcertsDBRepository concertsDbRepository;

        public ServiceConcert(ConcertsDBRepository concertsDbRepository)
        {
            this.concertsDbRepository = concertsDbRepository;
        }

        public IEnumerable<Concert> findAll() {
            return concertsDbRepository.FindAll();
        }

        public IEnumerable<Concert> getConcertsByDate(string date)
        {
            return concertsDbRepository.getConcertsByDate(date);
        }

        public void updateConcertTickets(int id, int numberOfTickets) {
            Concert concert = concertsDbRepository.FindOne(id);
            int newTicketsAvailable = concert.TicketsAvailable - numberOfTickets;
            int newTicketsSold = concert.TicketsSold + numberOfTickets;
            concertsDbRepository.UpdateConcertTickets(id, newTicketsAvailable, newTicketsSold);
        }

        public Concert getConcertByDateLocationStart(string date, string location, string startTime)
        {
            return concertsDbRepository.getConcertByDateLocationStart(date, location, startTime);
        }
        
    }
}